server/services/AssessmentDayDataService/index.js

```diff
@@ -0,0 +1,36 @@
+import deepEqual from 'deep-equal'
+
+class AssessmentDayDataService {
+  constructor(newAssessmentDayData, currentAssessmentDayData) {
+    this.newAssessmentDayData = newAssessmentDayData
+    this.currentAssessmentDayData = currentAssessmentDayData
+  }
+  filterNewData = () =>
+    this.newAssessmentDayData.filter(
+      ({ day }) =>
+        this.currentAssessmentDayData.findIndex((el) => el.day === day) === -1
+    )
+
+  filterUpdatedData = () => {
+    const filteredAssessments = this.newAssessmentDayData.reduce(
+      (listOfUpdates, participantData) => {
+        const isDayToBeUpdated = this.currentAssessmentDayData.findIndex(
+          (data) => data.day === participantData.day
+        )
+        if (isDayToBeUpdated !== -1) listOfUpdates.push(participantData)
+
+        return listOfUpdates
+      },
+      []
+    )
+
+    return filteredAssessments.filter((newDayData) => {
+      const currentData = this.currentAssessmentDayData.find(
+        (currentDayData) => currentDayData.day === newDayData.day
+      )
+      return !deepEqual(newDayData, currentData)
+    })
+  }
```

Comment:
Minor suggestion here that could clean up these methods a little. I know this is pre-existing code that we're moving to a new branch but wanted to point it out for future refactor opportunity.

```suggestion
  filterNewData = () => this.newAssessmentDayData.filter(({ day }) => !!this._currentDataForDay(day))

  filterUpdatedData = () => {
    return this.newAssessmentDayData.reduce(
      (newDataList, newDayData) => {
        const currentData = this._currentDataForDay(newDayData.day)
        const isDayToBeUpdated = !currentData || !deepEqual(newDayData, currentData)

        if (isDayToBeUpdated) newDataList.push(newDayData)

        return newDataList
      },
      []
    )
  }

  _currentDataForDay = (day) => this.currentAssessmentDayData.find((data) => data.day === day)
```